sds/src/proximity_keywords/mod.rs

```diff
@@ -439,9 +439,21 @@ pub enum ProximityKeywordsValidationError {
     #[error("Empty keywords are not allowed")]
     EmptyKeyword,
 
+    #[error("Duplicate keywords are not allowed ({0})")]
+    DuplicateKeyword(String),
+
+    #[error("`{0}` keyword starts or ends with a whitespace - spaces are only authorized in the middle of the keyword")]
+    KeywordStartsOrEndsWithSpace(String),
+
+    #[error("`{0}` keyword contains uppercase characters - matching is case-insensitive so using lower cased characters yields the same results")]
+    KeywordHasUppercase(String),
+
     #[error("Keywords cannot be longer than the look ahead character count ({0})")]
     KeywordTooLong(usize),
 
+    #[error("`{0}` keyword contains invalid characters - only alphanumeric characters and the following ones (` `, `@`, `.`, `\\`, `/`, `#`, `&`, `+`, `-`)")]
+    KeywordInvalidCharacters(String),
+
```

Comment:
@vinckama I am migrating the rest of the code to our internal library (sds-shared) - however I feel like this part is relevant to this library, as it is simply an extension of the coverage of `ProximityKeywordsValidationError`. This error is already in the library and had some corner cases it was lacking.

The core library uses it in `compile_keywords_to_ast`. Rules are compiled and the errors are generated when malformed rules are given. Perhaps the coverage of that function could be improved at some point and these new errors would be used there?

Reply:
Well turns out one can create rules in this package that violate these limitations described by these errors without any issue. I'll thus remove them from here and close the PR.